File: leetCode.com/typescript/trees/BinaryTreeLevelOrderTraversal.ts
// Given the root of a binary tree, return the level order traversal of its nodes' values. (i.e., from left to right, level by level).

function levelOrder(root: TreeNode | null): number[][] {
    if (!root)
        return []

    const result: number[][] = []
    let queue = [root]
    while (queue.length) {
        const level: number[] = []
        const nextQueue: TreeNode[] = []
        queue.forEach(n => {
            level.push(n.val)
            if (n.left)
                nextQueue.push(n.left)
            if (n.right)
                nextQueue.push(n.right)
        })
        queue = nextQueue
        result.push(level)
    }

    return result
};

File: leetCode.com/python/Trees/BinaryTreeLevelOrderTraversal.py
# Given the root of a binary tree, return the level order traversal of its nodes' values. (i.e., from left to right, level by level).

from typing import Optional
from classes import TreeNode


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> list[list[int]]:
        if not root:
            return []
        result = []
        queue = [root]
        while queue:
            level = []
            nextQueue = []
            for n in queue:
                level.append(n.val)
                if n.left:
                    nextQueue.append(n.left)
                if n.right:
                    nextQueue.append(n.right)
            result.append(level)
            queue = nextQueue
        return result